Simbody/src/CableSpan.cpp

```
@@ -0,0 +1,3028 @@
+        shootNewGeodesic(
+            getContactPointHint(),     // = location
+            nextPoint_S - prevPoint_S, // = tangent
+            0.,                        // = arc length
+            getDataInst(state).integratorInitialStepSize,
+            updDataInst(state));
+    }
+
+    // Helper function for detecting touchdown on the obstacle's surface, and
+    // the location of the touchdown point if so.
+    void calcCurveTouchdownIfNeeded(const State& state) const
+    {
+        // Only attempt touchdown when lifted.
+        const CurveSegmentData::Instance& dataInst = getDataInst(state);
+        if (dataInst.wrappingStatus !=
+            ObstacleWrappingStatus::LiftedFromSurface) {
+            return;
+        }
+
+        // Given the straight line between the point before and after this
+        // segment, touchdown is detected by computing the point on that line
+        // that is nearest to the surface.
+        const Vec3 prevPoint_S = findPrevPathPoint_S(state);
+        const Vec3 nextPoint_S = findNextPathPoint_S(state);
+
+        // Use the cached tracking point as the initial guess.
+        Vec3 pointOnLineNearSurface_S = dataInst.trackingPointOnLine_S;
+
+        // Compute the point on the line nearest the surface.
+        const IntegratorTolerances& tols = getIntegratorTolerances();
+        bool touchdownDetected           = false;
+        try {
+            // This routine might fail to converge if the line is too far from
+            // the surface. This should not be a problem, because in that case
+            // we definitely have not touched down.
+            touchdownDetected =
+                getContactGeometry().calcNearestPointOnLineImplicitly(
+                    prevPoint_S,
+                    nextPoint_S,
+                    tols.constraintProjectionMaxIterations,
+                    tols.constraintProjectionTolerance,
+                    pointOnLineNearSurface_S);
+        } catch (...) {
```

Pokemon exception catchers are usually a sign of a bad design *somewhere*. They cost a bunch to maintain and they will swallow up stuff that has no relationship to errors in your algorithm. Plus, how you're swallowing them makes them entirely invisible. The user has no way of knowing if (e.g.) there's a memory issues, out of bounds detected, etc.

---

fixed by https://github.com/simbody/simbody/pull/791#discussion_r1668415912